src/server/cluster_config_test.cc

```diff
@@ -126,6 +126,51 @@ TEST_F(ClusterConfigTest, ConfigSetMultipleInstances) {
   }
 }
 
+TEST_F(ClusterConfigTest, ConfigSetGetDeletedSlots) {
+  auto ds =
+      config_.SetConfig({{.slot_ranges = {{.start = 0, .end = 5'000}},
+                          .master = {.id = kMyId, .ip = "192.168.0.100", .port = 7000},
+                          .replicas = {}},
+                         {.slot_ranges = {{.start = 5'001, .end = 10'000}},
+                          .master = {.id = "other-master", .ip = "192.168.0.102", .port = 7002},
+                          .replicas = {}},
+                         {.slot_ranges = {{.start = 10'001, .end = 0x3FFF}},
+                          .master = {.id = "other-master2", .ip = "192.168.0.104", .port = 7004},
+                          .replicas = {}}});
+
+  EXPECT_TRUE(ds);
+  EXPECT_TRUE(ds.value().empty());  // On first config set no deleted slots.
+
+  ds = config_.SetConfig({{.slot_ranges = {{.start = 0, .end = 6'000}},
+                           .master = {.id = kMyId, .ip = "192.168.0.100", .port = 7000},
+                           .replicas = {}},
+                          {.slot_ranges = {{.start = 6'001, .end = 10'000}},
+                           .master = {.id = "other-master", .ip = "192.168.0.102", .port = 7002},
+                           .replicas = {}},
+                          {.slot_ranges = {{.start = 10'001, .end = 0x3FFF}},
+                           .master = {.id = "other-master2", .ip = "192.168.0.104", .port = 7004},
+                           .replicas = {}}});
+
+  EXPECT_TRUE(ds);
+  EXPECT_TRUE(ds.value().empty());  // On second config no slots taken from ownership
+
+  ds = config_.SetConfig({{.slot_ranges = {{.start = 0, .end = 5'997}},
+                           .master = {.id = kMyId, .ip = "192.168.0.100", .port = 7000},
+                           .replicas = {}},
+                          {.slot_ranges = {{.start = 5'998, .end = 10'000}},
+                           .master = {.id = "other-master", .ip = "192.168.0.102", .port = 7002},
+                           .replicas = {}},
+                          {.slot_ranges = {{.start = 10'001, .end = 0x3FFF}},
+                           .master = {.id = "other-master2", .ip = "192.168.0.104", .port = 7004},
+                           .replicas = {}}});
+
+  EXPECT_TRUE(ds);
+  EXPECT_FALSE(ds.value().empty());
+  EXPECT_TRUE(ds.value().contains(5'998));
+  EXPECT_TRUE(ds.value().contains(5'999));
+  EXPECT_TRUE(ds.value().contains(6'000));
```

Comment:
Did you try:

```cpp
EXPECT_THAT(ds.value(), UnorderedElementsAre(5'998, 5'999, 6'000));
```

If so, you could drop the `empty()` (or `size()`) expectation above